### ISSUE TITLE
Bug in JSString.isInstance()

### ISSUE THREAD

**Issue by salmonb** [konsoletyper/teavm#166]
I noticed a little bug in JSString.java

```java
@JSBody(params = "obj", script = "return typeof this === 'string';")
public static native boolean isInstance(JSObject obj);
```

It should be obj instead of this:

```java
@JSBody(params = "obj", script = "return typeof obj === 'string';")
public static native boolean isInstance(JSObject obj);
```

Also, would it be possible to have this isInstance() method implemented for other types?

**Comment by salmonb** [comment]
It is now fixed in TeaVM 0.4.1, so I'm closing this issue. Would be good however to have isInstance() also implemented in JSBoolean, JSNumber and eventually JSArray. Thank you.

**Comment by konsoletyper** [comment]
Oh, I see. Just forgot about this. May be I'll implement them in future versions.
